src/Datadog.Trace.AspNet/TracingHttpModule.cs

```diff
@@ -187,5 +188,31 @@ private void OnError(object sender, EventArgs eventArgs)
                 Log.Error(ex, "Datadog ASP.NET HttpModule instrumentation error");
             }
         }
+
+        private void SetStatusCode(Scope scope, int statusCode)
+        {
+            try
+            {
+                if (scope?.Span != null)
+                {
+                    scope.Span.SetTag(Tags.HttpStatusCode, statusCode.ToString());
+
+                    if (statusCode / 100 == 5)
```

Comment:
Should we move this logic into `Span.SetTag()`? This way the error flag gets sets no matter where the http status code is set from.

Reply:
I don't think this logic should move into the Span class. As much as we can, we should just store state in the Span, whereas the decision regarding status codes / error conditions are Tracer / Integration specific

Reply:
Ok, maybe not into the `Span` class, but maybe a helper method all the web integration can share instead of duplicating the logic?

Reply:
Yeah refactoring into one method would be good. I've done that in fc1825d4f57f0c123b29dbf73323f1de8e088b1a